### ISSUE TITLE
How to setup `nyx` to release with Gitlab CI/CD? 

### ISSUE THREAD

**Issue by sh-cau** [mooltiverse/nyx#291]
I want to have a scheduled release, say, each week using a Gitlab pipeline.
Unfortunately, there's no real example on how to do this.
I want `nyx` to take care of that by setting up my `.gitlab_ci.yml` as follows:
```yaml
image: 
  name: "mooltiverse/nyx:latest"
  entrypoint: [""]

variables:
  GIT_STRATEGY: clone
  GIT_DEPTH: "0"

stages:
  - release

release:
  stage: release
  tags:
    - docker
  rules:
    - if: '$CI_COMMIT_BRANCH=="master"'
  script:
    - git config --global user.name $GITLAB_USER_LOGIN
    - git config --global user.email $GITLAB_USER_EMAIL
    - nyx --debug publish
```
I had to include the `git config` bits because otherwise `nyx` complained that an `author` must be configured in order to push back to gitlab.

Unfortunately, I get the error 
```
time="2024-02-26T13:20:20Z" level=debug msg="attempting push to 'origin' using user name and password credentials."
time="2024-02-26T13:20:20Z" level=debug msg="pushing changes to remote repository 'origin' using username and password"
time="2024-02-26T13:20:20Z" level=debug msg="username and password authentication will use custom authentication options"
an error occurred when trying to push: authentication required
```
in the Gitlab pipeline with a gitlab runner.
I've set up the `.nyx.yaml` as described in #289, i.e., it should authenticate with a token `GITLAB_TOKEN` that I define in Gitlab and that is passed as an environment variable. It seems that `nyx` ignores that part in the `.nyx.yaml` file?
It is also possible that I am missing something here? 
I think the error message (or the debug messages) could be more enlightening, e.g., it would be nice to confirm that a TOKEN is used and not really a user-password combination.

**Comment by sh-cau** [comment]
Ok, I finally got it working. So there were two issues:
- Nyx did not infer the correct version, i.e., landed in a detached head state and then tried to push into nothingness. I solved this with the fix mentioned in #280, i.e., manually checking out the commit branch and resetting the repo. So this is also an issue for me.
-  I also had to apply the fix in #127, i.e., setting the repo URL manually, which is also quite annoying.
